src/io/inputs/data-series-csv-importer/DataSeriesRepoImporter.cpp

```diff
@@ -19,116 +19,149 @@
 #include <ranges>
 #include <vector>
 
-#include <boost/algorithm/string.hpp>
+#include <boost/iostreams/device/mapped_file.hpp>
+#include <boost/iostreams/stream.hpp>
 
 #include <antares/io/inputs/data-series-csv-importer/DataSeriesRepoImporter.h>
 #include <antares/optimisation/linear-problem-data-impl/timeSeriesSet.h>
+
 namespace fs = std::filesystem;
 
 namespace Antares::IO::Inputs::DataSeriesCsvImporter
 {
 using namespace Optimisation::LinearProblemDataImpl;
 
-static std::vector<double> readLine(std::string line,
-                                    std::string csvSeparators,
-                                    const std::string fileName)
+inline const char* parseOneDouble(const char* ptr,
+                                  const char* end,
+                                  double& value,
+                                  const std::string& errorMessagePrefix = "")
 {
-    std::vector<double> row;
-    std::vector<std::string> splitLine;
-    boost::split(splitLine, line, boost::is_any_of(csvSeparators));
-    std::transform(splitLine.begin(),
-                   splitLine.end(),
-                   std::inserter(row, row.end()),
-                   [fileName](std::string& s)
-                   {
-                       boost::trim(s);
-                       if (s.empty())
-                       {
-                           throw std::invalid_argument(
-                             fileName + ": columns have inconsistent number of rows");
-                       }
-                       try
-                       {
-                           return std::stod(s);
-                       }
-                       catch (const std::invalid_argument&)
-                       {
-                           throw std::invalid_argument(fileName + ": \"" + s
-                                                       + "\" is not a number");
-                       }
-                   });
-    return row;
+    auto [p, ec] = std::from_chars(ptr, end, value);
+    if (ec == std::errc::invalid_argument)
+    {
+        throw std::invalid_argument(errorMessagePrefix + ": \"" + *p + "\" is not a number");
+    }
+    return p;
 }
 
-static std::vector<std::vector<double>> csvToMatrix(const std::filesystem::path& path,
-                                                    std::string csvSeparators)
+static std::vector<double> parseRow(const char* first,
+                                    const char* last,
+                                    char sep = ' ',
+                                    size_t capacity = 0,
+                                    const std::string& errorMessagePrefix = "")
 {
-    std::vector<std::vector<double>> result;
-    std::ifstream infile(path, std::ios_base::binary | std::ios_base::in);
-    if (!infile.is_open())
+    std::vector<double> row;
+    if (capacity > 0)
     {
-        throw std::invalid_argument("Could not open file " + path.filename().string());
+        row.reserve(capacity);
     }
```

Comment:
If `capacity=0`, this is a no-op.
```suggestion
    row.reserve(capacity);
```